AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/UserPoolConfigurationData.swift

```diff
@@ -83,3 +100,136 @@ extension UserPoolConfigurationData.CustomEndpoint {
         validatedHost = endpoint.host
     }
 }
+
+extension UserPoolConfigurationData {
+
+    /// settings used in the Authenticator
+    struct PasswordProtectionSettings: Equatable, Codable {
```

Comment:
In the current PR, we will translate the object back to the `@_spi` backed `jsonConfiguration` property on the plugin that Authenticator uses https://github.com/aws-amplify/amplify-swift/blob/main/AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin.swift#L39

To answer your question, Authenticator will later use these types instead of the `JSONValue`, but when we do this, we will still use `@_spi`, so these types won't be public

Reply:
Aah my bad, I thought we were putting the migration into `AuthConfiguration` into this work as well.
But got it, the Authenticator will keep using `jsonConfiguration` for now, which I see is being manually built to look like the old one :) 👍